### PR TITLE
~subsquid~ to => firesquid

### PR DESCRIPTION
**Pull request by easteregg**
## Issue
https://app.clickup.com/t/2tfg1q3

## Description
This PR fixes the build issue with subsquid and removal of `npx squid-graphql-server`. as well as migrates to firesquid as per https://docs.subsquid.io/migrate/migrate-to-fire-squid/

## Checklist

- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_
- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: subsquid/src/bondedFinanceProcessor.ts [hunk 1]
```diff
@@ -22,18 +23,18 @@ interface NewBondEvent {
...
 function getNewOfferEvent(event: BondedFinanceNewOfferEvent): NewOfferEvent {
-  const { offerId, beneficiary } = event.asV2401 ?? event.asLatest;
+  const { offerId, beneficiary } = event.asV2401;
 
   return { offerId, beneficiary };
 }
 
 export async function processNewOfferEvent(
-  ctx: EventHandlerContext,
+  ctx: EventHandlerContext<Store, { event: true }>,
```
Comment (andresmechali) [inline subsquid/src/bondedFinanceProcessor.ts:37]: is this `event: true` needed?

**Review by andresmechali — APPROVED** [review]

**Review by cor — CHANGES_REQUESTED** [review]
Let's also include the nixification in this PR and mark it as draft for now so that it does not get merged

File: subsquid/src/bondedFinanceProcessor.ts (hunk 1 above)
Comment (easteregg) [reply subsquid/src/bondedFinanceProcessor.ts:37]: I have kept it as a placeholder for future needs:
https://docs.subsquid.io/migrate/migrate-to-fire-squid/#3-change-the-processor-initialization

Not sure what to put as a replacement, happy to do it if you have any suggestion that leads me to the right direction 😄

**Comment by easteregg** [comment]
> Let's also include the nixification in this PR and mark it as draft for now so that it does not get merged

Converted to draft.

**Review by vimukthi-git — APPROVED** [review]
LGTM, did not run locally though

**Review by saadjhk — APPROVED** [review]

**Review by cor — APPROVED** [review]
LGTM

**Comment by cor** [comment]
@mergify refresh

**Comment by cor** [comment]
@mergify update

**Pull request merged**